fe/parameterize_batch_thermostat.py

```diff
@@ -0,0 +1,440 @@
+    parser.add_argument('--num_trials', type=int, required=True)
+    parser.add_argument('--forcefield', type=str, required=True)
+    args = parser.parse_args()
+
+    assert os.path.isdir(args.out_dir)
+
+    if args.precision == 'single':
+        precision = np.float32
+    elif args.precision == 'double':
+        precision = np.float64
+    else:
+        raise Exception("precision must be either single or double")
+
+    num_gpus = args.num_gpus
+    all_du_dls = []    
+
+    host_pdb_file = args.complex_pdb
+    host_pdb = app.PDBFile(host_pdb_file)
+    host_conf = []
+    for x,y,z in host_pdb.positions:
+        host_conf.append([to_md_units(x),to_md_units(y),to_md_units(z)])
+    host_conf = np.array(host_conf)
+    host_name = "complex"
+
+    # set up the system
+    amber_ff = app.ForceField('amber99sbildn.xml', 'amber99_obc.xml')
+    host_system = amber_ff.createSystem(host_pdb.topology,
+        nonbondedMethod=app.NoCutoff,
+        constraints=None,
+        rigidWater=False)
+
+    cutoff = 1.25
+
+    host_system = openmm_converter.deserialize_system(host_system, cutoff=cutoff)
+    num_host_atoms = len(host_system.masses)
+
+    print("num_host_atoms", num_host_atoms)
+
+    for ligand_sdf_file in args.ligand_sdf:
+        for guest_mol in Chem.SDMolSupplier(ligand_sdf_file, removeHs=False):
+            break
+
+    open_ff = forcefield.Forcefield(args.forcefield)
+    nrg_fns = open_ff.parameterize(guest_mol, cutoff=cutoff)
+    guest_masses = get_masses(guest_mol)
+    guest_system = system.System(nrg_fns, open_ff.params, open_ff.param_groups, guest_masses)
+
+    combined_system = host_system.merge(guest_system)
+
+    temperature = 300
+    dt = 1.5e-3
+    friction = 91
+
+    ca, cbs, ccs = langevin_coefficients(
+        temperature,
+        dt,
+        friction,
+        np.array(combined_system.masses)
+    )
+
+    cbs *= -1
+
+    print("Integrator coefficients:")
+    print("ca", ca)
+    print("cbs", cbs)
+    print("ccs", ccs)
+
+    # insertion only
+    Ts = [
+        1000, # fast insertion
+        5000, # slow insertion
+    ]
+
+    complete_T = np.sum(Ts)
+
+    complete_lambda = np.concatenate([
+        np.linspace(cutoff, 0.3, Ts[0]),
+        np.linspace(0.3,   0.0,  Ts[1]),
+    ])
+
+    complete_cas = np.concatenate([
+        np.linspace(0.00, ca, Ts[0]),
+        np.linspace(ca, ca, Ts[1]),
+    ])
+
+    complete_dts = np.concatenate([
+        np.linspace(dt, dt, Ts[0]),
+        np.linspace(dt, dt, Ts[1]),
+    ])
+
+    lambda_idxs = np.zeros(len(combined_system.masses), dtype=np.int32)
+    lambda_idxs[num_host_atoms:] = 1
+
+    sim = simulation.Simulation(
+        combined_system,
+        complete_dts,
+        complete_cas,
+        cbs,
+        ccs,
+        complete_lambda,
+        lambda_idxs,
+        precision
+    )
+
+    initial_params = sim.system.params
+
+    lr = 1e-4
+    opt_init, opt_update, get_params = optimizers.adam(lr)
```

Comment:
lr=1e-4 is probably too small for adam

Reply:
you can test this with a fixed seed and seed if adam can learn anything

Reply:
Got it!